lib/market.rb

```diff
@@ -0,0 +1,26 @@
+class Market
+    attr_reader :name, :vendors
+    def initialize(name)
+        @name = name
+        @vendors = []
+    end
+
+    def add_vendor(vendor)
+        @vendors << vendor
+    end
+
+    def vendor_names
+        @vendors.map(&:name)
+    end
+
+    def vendors_that_sell(item)
+        vendor_items = []
+        @vendors.each do |item|
+            if @vendors.include?(item.name)
```

Comment:
The key here is that you need to use code youve already written. Instead of using include here, use the check_stock method youve already got.